src/accessory/HumidifierAccessory.ts

```diff
@@ -0,0 +1,105 @@
+import { PlatformAccessory } from 'homebridge';
+import { TuyaPlatform } from '../platform';
+import BaseAccessory from './BaseAccessory';
+
+export default class HumidifierAccessory extends BaseAccessory {
+
+  constructor(platform: TuyaPlatform, accessory: PlatformAccessory) {
+    super(platform, accessory);
+
+    this.configureActive();
+    this.configureTargetState();
+    this.configureCurrentState();
+    this.configureCurrentRelativeHumidity();
+    this.configureRelativeHumidityHumidifierThreshold();
+    this.configureTemperatureSensor();
+  }
+
+  mainService() {
+    return this.accessory.getService(this.Service.HumidifierDehumidifier)
+      || this.accessory.addService(this.Service.HumidifierDehumidifier);
+  }
+
+  configureActive(){
+    const { ACTIVE, INACTIVE } = this.Characteristic.Active;
+    this.mainService().getCharacteristic(this.Characteristic.Active)
+      .onGet(() => {
+        const status = this.getStatus('switch');
+        return (status?.value as boolean) ? ACTIVE : INACTIVE;
+      })
+      .onSet(value => {
+        this.sendCommands([{ code: 'switch', value: (value === ACTIVE) ? true : false }]);
+      });
+  }
+
+  configureTargetState(){
+    const { HUMIDIFIER} = this.Characteristic.TargetHumidifierDehumidifierState;
+    const validValues = [ HUMIDIFIER ];
+
+    this.mainService().getCharacteristic(this.Characteristic.TargetHumidifierDehumidifierState)
+      .onGet(() => {
+        return HUMIDIFIER;
+      }).setProps({ validValues });
+  }
+
+  configureCurrentState(){
+    const { INACTIVE, HUMIDIFYING } = this.Characteristic.CurrentHumidifierDehumidifierState;
+
+    this.mainService().getCharacteristic(this.Characteristic.CurrentHumidifierDehumidifierState)
+      .onGet(() => {
+        const status = this.getStatus('switch');
+        return (status?.value as boolean) ? HUMIDIFYING : INACTIVE;
+      });
+  }
+
+  configureCurrentRelativeHumidity(){
+    this.mainService().getCharacteristic(this.Characteristic.CurrentRelativeHumidity)
+      .onGet(() => {
+        const status = this.getStatus('humidity_current');
+        const humidity = Math.min(100, Math.max(0, status?.value as number));
+        return humidity;
+      });
+  }
+
+  configureRelativeHumidityHumidifierThreshold(){
+    const schema = this.getSchema('humidity_set');
+    const s = JSON.parse(schema?.values as string);
+    if (!schema) {
+      return;
+    }
+
+    this.mainService().getCharacteristic(this.Characteristic.RelativeHumidityHumidifierThreshold)
+      .onGet(() => {
+        const status = this.getStatus('humidity_set');
+        let humidity_set = Math.max(0, status?.value as number);
```

Comment:
Accroding to the different devices, humidity value may not in the range `0~100`, maybe `0~1000`, and we should process the `schema.property.scale`, `scale=0` means `min` and `max` need to divided by `10^0`, `scale=1` then divided by `10^1`. These codes can be found in the `TemperatureHumiditySensorAccessory`.

Reply:
I wasn't aware, sorry :) Fixed